ToNote/Models/ToDo.cs

```diff
@@ -35,5 +45,32 @@ public int Index
                 }
             }
         }
+
+        private DateTime? _SelectedDate;
+
+        public DateTime? SelectedDate
+        {
+            get => _SelectedDate;
+            set
+            {
+                if (_SelectedDate != value)
+                {
+                    _SelectedDate = value;
+
+                    RaisePropertyChanged(nameof(SelectedDate));
+                    RaisePropertyChanged(nameof(IsDatePast));
+                }
+                
+            }
+        }
+
+        private DateTime? _LastDone;
```

Comment:
As you don't raise any events here on property changed, you could just make this `public DateTime? LastDone { get; private set; }`